libs/contrib/Data/Nat/Exponentiation.idr

```diff
@@ -0,0 +1,83 @@
+module Data.Nat.Exponentiation
+
+import Data.Nat as Nat
+import Data.Monoid.Exponentiation as Mon
+import Data.Num.Implementations as Num
+import Data.Nat.Views
+import Data.Nat.Order
+import Decidable.Order
+import Syntax.PreorderReasoning
+import Syntax.PreorderReasoning.Generic
+
+%default total
+
+public export
+pow : Nat -> Nat -> Nat
+pow = Mon.(^)
```

Comment:
Now we have `pow` in `contrib` and `power` in `prelude` that are practically the same, right?

Reply:
I did not know about `power`. This one is (runtime) efficient.